generator/generator.cpp

```diff
@@ -1,36 +1,49 @@
 
 
-#include "generator.h"
+#include "generator.hpp"
 
-UDPGenerator::UDPGenerator() = default;
+UDPGenerator::UDPGenerator() : UDPGenerator(0){}
 
-UDPGenerator::~UDPGenerator() = default;
-
-void UDPGenerator::startGenerator(unsigned short sport, std::string d_ip, unsigned short dport) {
+UDPGenerator::UDPGenerator(unsigned short sport) {
     struct sockaddr_in s_addr;
+    t_running = true;
 
-    try {
-        if((this_socket = socket(AF_INET, SOCK_DGRAM, 0)) < 0)
-            throw "socket function";
-        //errno
-        s_addr.sin_family = AF_INET;
-        s_addr.sin_port = htons(sport);
-        s_addr.sin_addr.s_addr = htonl(INADDR_ANY);
+    if((this_socket = socket(AF_INET, SOCK_DGRAM, 0)) < 0)
+        throw std::runtime_error("error create socket!");
 
-        if(bind(this_socket, (struct sockaddr *)&s_addr, sizeof(s_addr)) < 0) {
-            throw "bind function";
+    s_addr.sin_family = AF_INET;
+    s_addr.sin_port = htons(sport);
+    s_addr.sin_addr.s_addr = htonl(INADDR_ANY);
 
-        }
-        struct sockaddr_in d_addr;
-        d_addr.sin_family = AF_INET;
-        inet_pton(AF_INET, d_ip.c_str(), &(d_addr.sin_addr));
-        d_addr.sin_port = htons(dport);
-        handle(d_addr);
-    }
-    catch (const char * err) {
-        perror(err);
-        closeGenerator();
+    if(bind(this_socket, (struct sockaddr *)&s_addr, sizeof(s_addr)) < 0)
+        throw std::runtime_error("error bind socket with address!");
+}
+
+
+UDPGenerator::~UDPGenerator() {
+    close(this_socket);
+}
+void UDPGenerator::setDestAddr(std::string dest_ip, unsigned short dport) {
+    d_addr.sin_family = AF_INET;
+    inet_pton(AF_INET, dest_ip.c_str(), &(d_addr.sin_addr));
+    d_addr.sin_port = htons(dport);
+}
+
+void UDPGenerator::startGenerator() {
+    boost::thread sender(&UDPGenerator::handle, this);
```

Comment:
проще лямбдой написать
```suggestion
    std::thread sender([&] {handle()});
```